File: tests/test_user.py
import unittest
from usecases.user_usecase import UserUseCase
from usecases.merchant_usecase import MerchantUseCase
from datastore.local_datastore import LocalDataStore


class TestUser(unittest.TestCase):
    _user_use_case = UserUseCase(LocalDataStore())
    _merchant_use_case = MerchantUseCase(LocalDataStore())

    def setUp(self):
        self._merchant_use_case.new_merchant('m1', 'm1@email.com', '0.5%')
        self._merchant_use_case.new_merchant('m2', 'm2@email.com', '1.5%')
        self._merchant_use_case.new_merchant('m3', 'm3@email.com', '1.25%')

    def test_1(self):
        """New User"""
        tests = [
            ('user1', 'user1@users.com', '300', True),
            ('user1', 'user2@users.com', '400', False),
            ('user2', 'user2@users.com', '400', True),
            ('user3', 'user3@users.com', '500', True)
        ]

        for test in tests:
            result = self._user_use_case.new_user(test[0], test[1], test[2])
            self.assertEqual(result['success'], test[3])

    def test_2(self):
        """Transaction"""
        tests = [
            ('user2', 'm1', '500', False),
            ('user1', 'm2', '300', True),
        ]
        for test in tests:
            result = self._user_use_case.transaction(test[0], test[1], test[2])
            self.assertEqual(result['success'], test[3])

    def test_3(self):
        """Payback Dues"""
        tests = [
            ('user1', '200', True),
            ('user1', '200', False)
        ]

        for test in tests:
            result = self._user_use_case.payback_dues(test[0], test[1])
            self.assertEqual(result['success'], test[2])

    def test_4(self):
        """Report Total Dues"""
        result = self._user_use_case.report_total_dues()
        self.assertEqual(result['success'], True)

    def test_5(self):
        """Report Users at Credit Limit"""
        result = self._user_use_case.report_users_at_credit_limit()
        self.assertEqual(result['success'], True)

    def test_6(self):
        """Report Dues"""
        tests = [
            ('user1', True),
            ('user4', False)
        ]
        for test in tests:
            result = self._user_use_case.report_dues(test[0])
            self.assertEqual(result['success'], test[1])

File: java/src/main/java/com/paylater/tests/TestUser.java
package com.paylater.tests;

import com.paylater.repository.LocalDatabase;
import com.paylater.usecases.MerchantUseCase;
import com.paylater.usecases.UserUseCase;
import com.paylater.utils.ResponseValue;
import junit.framework.TestCase;

public class TestUser extends TestCase {
    private final UserUseCase userUseCase = new UserUseCase(LocalDatabase.getInstance());
    private final MerchantUseCase merchantUseCase = new MerchantUseCase(LocalDatabase.getInstance());

    @Override
    public void setUp() {
        merchantUseCase.newMerchant("m1", "m1@email.com", "0.5%");
        merchantUseCase.newMerchant("m2", "m2@email.com", "1.5%");
        merchantUseCase.newMerchant("m3", "m3@email.com", "1.25%");
    }

    // New User
    public void test1() {
        System.out.println("Running New User");

        ResponseValue result1 = userUseCase.newUser("user1", "user1@email.com", 300);
        assertTrue(result1.isSuccess());

        ResponseValue result2 = userUseCase.newUser("user1", "user1@email.com", 400);
        assertFalse(result2.isSuccess());

        ResponseValue result3 = userUseCase.newUser("user2", "user2@email.com", 400);
        assertTrue(result3.isSuccess());

        ResponseValue result4 = userUseCase.newUser("user3", "user3@email.com", 500);
        assertTrue(result4.isSuccess());
    }

    // Transaction
    public void test2() {
        System.out.println("Running Transaction");

        ResponseValue result1 = userUseCase.transaction("user2", "m1", "500");
        assertFalse(result1.isSuccess());

        ResponseValue result2 = userUseCase.transaction("user1", "m2", "300");
        assertTrue(result2.isSuccess());
    }

    // Payback Dues
    public void test3() {
        System.out.println("Running Payback Dues");

        ResponseValue result1 = userUseCase.paybackDues("user1", "200");
        assertTrue(result1.isSuccess());

        ResponseValue result2 = userUseCase.paybackDues("user1", "200");
        assertFalse(result1.isSuccess());
    }

    // Report Total Dues
    public void test4() {
        System.out.println("Running Report Total Dues");

        ResponseValue result = userUseCase.reportTotalDues();
        assertTrue(result.isSuccess());
    }

    // Report Users At Credit Limit
    public void test5() {
        System.out.println("Running Report Users at Credit Limit");

        ResponseValue result = userUseCase.reportUsersAtCreditLimit();
        assertTrue(result.isSuccess());
    }

    // Report Dues
    public void test6() {
        System.out.println("Running Report Dues");

        ResponseValue result1 = userUseCase.reportDues("user1");
        assertTrue(result1.isSuccess());

        ResponseValue result2 = userUseCase.reportDues("user4");
        assertFalse(result2.isSuccess());
    }
}
